### PR TITLE
[onert] Introduce CircleExporter class

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit introduces CircleExporter class to export circle and circleplus file. It also adds an object-based API of the circle schema to generate a new circle file.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Draft: #13014

### CODE REVIEW

**Comment by hseok-oh** [comment]
Is there any difference between `runtime/libs/circle-schema/include/circle_schema_generated.h` and added generated file?

File: runtime/onert/core/src/exporter/CircleExporter.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    throw std::runtime_error("Failed to load original circle file");
+  _model = model->UnPack();
+}
+
+CircleExporter::~CircleExporter() { finish(); }
+
+void CircleExporter::finish()
+{
+  flatbuffers::FlatBufferBuilder builder(1024);
+  builder.Finish(::circle::Model::Pack(builder, _model), ::circle::ModelIdentifier());
```
Comment (ragmani) [inline runtime/onert/core/src/exporter/CircleExporter.cc:60]: Isn't there any memory leak?
How about passing address of ModelT object like below?
```suggestion
  builder.Finish(::circle::Model::Pack(builder, &_model), ::circle::ModelIdentifier());
```

- `CircleExporter`'s a data member for `ModelT`
```c++
::circle::ModelT _model;
```

- `Unpack`
```c++
model->UnPackTo(&_model);
```

**Comment by jyoungyun** [comment]
@hseok-oh 

> Is there any difference between `runtime/libs/circle-schema/include/circle_schema_generated.h` and added generated file? Please update `runtime/libs/circle-schema/include/circle_schema_generated.h` file instead of adding duplicated code.

This file is different from the `runtime/libs/circle-schema/include/circle_schema_generated.h` file. It is an object-based API file for creating a new circle with given circle file data.

**Comment by hseok-oh** [comment]
> @hseok-oh
> 
> > Is there any difference between `runtime/libs/circle-schema/include/circle_schema_generated.h` and added generated file? Please update `runtime/libs/circle-schema/include/circle_schema_generated.h` file instead of adding duplicated code.
> 
> This file is different from the `runtime/libs/circle-schema/include/circle_schema_generated.h` file. It is an object-based API file for creating a new circle with given circle file data.
> 
> You can see this comment at the top of the `circle_schema_generated.h` file.
> 
> ```
> // command: flatc -c --gen-onefile --gen-object-api ../../../../nnpackage/schema/circle_schema.fbs
> ```

Then please update  `runtime/libs/circle-schema/include/circle_schema_generated.h` as I commented.

File: runtime/onert/core/src/exporter/CircleExporter.cc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+CircleExporter::~CircleExporter() { finish(); }
+
+void CircleExporter::finish()
+{
+  flatbuffers::FlatBufferBuilder builder(1024);
+  builder.Finish(::circle::Model::Pack(builder, _model.get()), ::circle::ModelIdentifier());
+
+  std::ofstream dst(_path.c_str(), std::ios::binary);
+  dst.write((const char *)builder.GetBufferPointer(), builder.GetSize());
```
Comment (ragmani) [inline runtime/onert/core/src/exporter/CircleExporter.cc:63]: (Optional) How about using `static_cast` for clear and safe casting?
```suggestion
  dst.write(static_cast<const char *>(builder.GetBufferPointer()), builder.GetSize());
```

**Review by ragmani — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]

**Pull request merged**